File: Java/TP3_MET_JAVA/src/Inductif/MotVideInductif.java
package Inductif;

import Interface.Mot;

public enum MotVideInductif implements Mot {

    SINGLETON;

    public boolean casVide() {
        return true;
    }

    public Character caractere() {
        throw new UnsupportedOperationException();
    }

    public Mot reste() {
        throw new UnsupportedOperationException();
    }

    @Override
    public Mot creerCaractereMot(char c) {
        return new CaractereMotInductif(c, this);
    }

    @Override
    public Mot creerMotMot(Mot droite) {
        return new MotMotInductif(this, droite);
    }

    @Override
    public Mot creerMotVide() {
        return MotVideInductif.SINGLETON;
    }
}

File: TypeScript/TP3_MET_TS/src/Inductif/MotVideInductif.ts
import {CaractereMotInductif} from "./CaractereMotInductif";
import {MotMotInductif} from "./MotMotInductif";
import {Mot} from "../Interface/Mot";

export class MotVideInductif extends Mot {

    public static SINGLETON: MotVideInductif = new MotVideInductif();

    public casVide(): boolean {
        return true;
    }

    public creerCaractereMot(c: string): Mot {
        return new CaractereMotInductif(c, this);
    }

    public creerMotMot(droite: Mot): Mot {
        return new MotMotInductif(this, droite);
    }

    public creerMotVide(): Mot {
        return MotVideInductif.SINGLETON;
    }
}
